### PR TITLE
feature/oktane 2077

### PR DESCRIPTION
**Pull request by Joshua-Brent-Collins**
This will provide Okta authentication into spring secured grails applications.

### CODE REVIEW

**Comment by Joshua-Brent-Collins** [comment]
@eric-gonzales @dbpfindexp Much authentication, many protections 🐕

File: grails-app/init/org/grails/plugins/okta/authenticator/Application.groovy [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import grails.boot.*
+import grails.boot.config.GrailsAutoConfiguration
+import grails.plugins.metadata.*
+
+@PluginSource
+class Application extends GrailsAutoConfiguration {
+    static void main(String[] args) {
+        GrailsApp.run(Application, args)
+    }
+}
```
Comment (eric-gonzales) [inline grails-app/init/org/grails/plugins/okta/authenticator/Application.groovy]: newline

File: src/main/groovy/org/grails/plugins/okta/authenticator/OktaAuthUtil.groovy [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+import groovyx.net.http.ContentType
+import groovyx.net.http.Method
+import groovyx.net.http.RESTClient
+import groovyx.net.http.Status
+import org.apache.http.HttpStatus
+
+public class OktaAuthUtil {
+
+    def final OKTA_API_TOKEN = System.getenv('OKTA_API_TOKEN')
```
Comment (dbpfindexp) [inline src/main/groovy/org/grails/plugins/okta/authenticator/OktaAuthUtil.groovy]: Does `System.getenv` work on AWS?  I forget which one works which one doesn't.

**Comment by eric-gonzales** [comment]
@dbpfindexp - I'm good with this if you want to merge it.

File: src/main/groovy/org/grails/plugins/okta/authenticator/OktaAuthenticatorGrailsPlugin.groovy [hunk 3]
```diff
@@ -0,0 +1,90 @@
+package org.grails.plugins.okta.authenticator
+
+import grails.plugins.*
+import org.springframework.boot.web.servlet.FilterRegistrationBean
+
+class OktaAuthenticatorGrailsPlugin extends Plugin {
+
+    // the version or versions of Grails the plugin is designed for
+    def grailsVersion = "3.2.11 > *"
```
Comment (dbpfindexp) [inline src/main/groovy/org/grails/plugins/okta/authenticator/OktaAuthenticatorGrailsPlugin.groovy]: Does this need to be `>=`?

Comment (Joshua-Brent-Collins) [reply src/main/groovy/org/grails/plugins/okta/authenticator/OktaAuthenticatorGrailsPlugin.groovy]: Not sure this was auto generated

**Pull request merged**